src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamServiceAccountProvider.scala

```diff
@@ -17,7 +20,11 @@ abstract class SamServiceAccountProvider(config: Config) extends ServiceAccountP
   implicit val materializer = ActorMaterializer()
   implicit val executionContext = system.dispatcher
 
-  protected lazy val samConfig = config.as[SamConfig]("sam")
-  protected lazy val samDAO = new HttpSamDAO(samConfig.server)
+  protected lazy val samServer = config.as[String]("samServer")
```

Comment:
Just thinking since this is duplicated with code in `SamAuthProvider`, maybe there is an opportunity to standardize this into a trait which can be used by all providers. For example:

```
trait SamProvider {
  val config: Config

  protected lazy val samClient = new SwaggerSamClient(...)
}
```

```
class PetsPerProjectServiceAccountProvider(config: Config) extends ServiceAccountProvider(config) with SamProvider {
  // use samClient
}
```

```
class SamAuthProvider(authConfig: Config, serviceAccountProvider: ServiceAccountProvider) extends LeoAuthProvider(authConfig, serviceAccountProvider) with SamProvider {
  // use samClient
}
```

Reply:
Faces: this posed some problems, not going to do it